driver/driver.hpp

```diff
@@ -157,6 +160,140 @@ struct GPUMem
 #endif
 };
 
+template <typename Tgpu>
+class GpumemTensor
```

Comment:
Can this class be exposed in include/miopen? So we can include it in our Fin code? Otherwise, I think we are reinventing the wheel here: https://github.com/ROCm/MIFin/blob/develop/src/include/tensor.hpp

Reply:
@CAHEK7  thoughts?

Reply:
It should be in the next PR. Lets separate unrelated changes.